src/js/files/components/UploadItem.js

```diff
@@ -39,3 +41,42 @@ export const UploadItem = ({ name, progress, size }) => (
         </UploadItemTitle>
     </StyledUploadItem>
 );
+
+const StyledUploadItemError = styled(StyledUploadItem)`
+    i.fa-times {
+        color: ${props => getColor({ theme: props.theme, color: "red" })};
+        font-size: 20px;
+    }
+    span: last-child {
+        color: ${props => getColor({ theme: props.theme, color: "red" })};
+        font-size: 12px;
+        font-style: italic;
+    }
+    i.fa-trash {
+        margin-left: 5px;
+        font-size: 14px;
+    }
+`;
+
+export const UploadItemError = ({ name, onRemove, localId }) => (
+    <StyledUploadItemError>
+        <AffixedProgressBar now={100} color="red" />
+        <UploadItemTitle>
+            {<Icon name="times" />}
+            {name}
+            <span>
+                Failed <Icon aria-label={`delete ${name}`} name="trash" color="red" onClick={() => onRemove(localId)} />
+            </span>
+        </UploadItemTitle>
+    </StyledUploadItemError>
+);
+
+const mapDispatchToProps = dispatch => {
+    return {
+        onRemove: localId => {
+            dispatch(removeUpload(localId));
+        }
+    };
+};
+
+export const ConnectedUploadItemError = connect(null, mapDispatchToProps)(UploadItemError);
```

Comment:
Combine upload item implementations into single connected component and:
```js
export default connect(mapDispatchToProps)(UploadItem);
```

Reply:
The two are now combined per earlier suggestion. Also per the suggest line of code, I assume the main point was the removal of the null positional argument from: 

```javascript 
export default connect(null, mapDispatchToProps)(UploadItem);
```
As annoying as it looks, the null is there intentionally so that the first positional argument slot is occupied and `mapDispatchToProps` gets handed dispatch as expected. Written the suggested way, `mapDispatchToProps` is handed state.

Reply:
Ah makes sense, I had deliberately avoided it in the first push since it seemed strange to export the error variant of `UploadItem` as default. Once merged, I went back to exporting as default.